app/controllers/licence_controller.rb

```diff
@@ -37,15 +37,19 @@ def search
     @publication = publication
     @interaction_details = licence_details
 
-    @postcode = postcode if postcode_search_submitted?
-    @location_error = location_error if postcode_search_submitted?
-
-    if postcode_search_submitted? && local_authority_slug
-      redirect_to licence_authority_path(slug: params[:slug], authority_slug: local_authority_slug)
-    elsif authority_choice_submitted?
-      redirect_to licence_authority_path(slug: params[:slug], authority_slug: CGI.escape(params[:authority][:slug]))
+    if request.post?
```

Comment:
@whoojemaflip Do you think this makes it more clear in terms of what's happening when a POST happens?